panic: runtime error: invalid memory address or nil pointer dereference

**What happened**:

```
$ k -n shoot--it--tmk63-q6p get po cluster-autoscaler-8b4c75dd8-cbrrb
NAME                                 READY   STATUS   RESTARTS   AGE
cluster-autoscaler-8b4c75dd8-cbrrb   0/1     Error    7          12m
```

```
$ k -n shoot--it--tmk63-q6p logs cluster-autoscaler-8b4c75dd8-cbrrb -p

I0511 08:17:50.779956       1 main.go:277] Registered cleanup signal handler
I0511 08:17:50.779994       1 node_instances_cache.go:156] Start refreshing cloud provider node instances cache
I0511 08:17:50.780026       1 node_instances_cache.go:168] Refresh cloud provider node instances cache finished, refresh took 26.401µs
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2a3de5b]

goroutine 114 [running]:
k8s.io/autoscaler/cluster-autoscaler/cloudprovider/mcm.(*McmManager).GetMachineDeploymentNodeTemplate(0xc000bb56c0, 0xc0007ba300, 0xc000bb56c0, 0xc000581800, 0x7ff5670d9108)
	/tmp/build/80754af9/git-gardener.autoscaler-machine-controller-manager-provider.machine-controller-manager-provider/tmp/src/k8s.io/autoscaler/cluster-autoscaler/cloudprovider/mcm/mcm_manager.go:687 +0xb9b
k8s.io/autoscaler/cluster-autoscaler/cloudprovider/mcm.(*MachineDeployment).TemplateNodeInfo(0xc0007ba300, 0x7ffc8cf14a9c, 0x1c, 0xc001331160)
	/tmp/build/80754af9/git-gardener.autoscaler-machine-controller-manager-provider.machine-controller-manager-provider/tmp/src/k8s.io/autoscaler/cluster-autoscaler/cloudprovider/mcm/mcm_cloud_provider.go:386 +0x38
k8s.io/autoscaler/cluster-autoscaler/core/utils.GetNodeInfoFromTemplate(0x3c47ea0, 0xc0007ba300, 0xc00092c400, 0x5, 0x8, 0x3bde2e0, 0xc000a65340, 0xc000a9c4b0, 0x0, 0x0, ...)
	/tmp/build/80754af9/git-gardener.autoscaler-machine-controller-manager-provider.machine-controller-manager-provider/tmp/src/k8s.io/autoscaler/cluster-autoscaler/core/utils/utils.go:170 +0x7d
k8s.io/autoscaler/cluster-autoscaler/core/utils.GetNodeInfosForGroups(0xc0009ae650, 0x2, 0x2, 0xc000a9c810, 0x3c3fc80, 0xc000a9c330, 0x3c3ba00, 0xc000e60000, 0xc00092c400, 0x5, ...)
	/tmp/build/80754af9/git-gardener.autoscaler-machine-controller-manager-provider.machine-controller-manager-provider/tmp/src/k8s.io/autoscaler/cluster-autoscaler/core/utils/utils.go:114 +0x8b2
k8s.io/autoscaler/cluster-autoscaler/core.(*StaticAutoscaler).RunOnce(0xc001302b40, 0xc01ead4e2e80a308, 0x27314e48b, 0x5653ca0, 0x0, 0x0)
	/tmp/build/80754af9/git-gardener.autoscaler-machine-controller-manager-provider.machine-controller-manager-provider/tmp/src/k8s.io/autoscaler/cluster-autoscaler/core/static_autoscaler.go:268 +0x994
main.run(0xc00077e230)
	/tmp/build/80754af9/git-gardener.autoscaler-machine-controller-manager-provider.machine-controller-manager-provider/tmp/src/k8s.io/autoscaler/cluster-autoscaler/main.go:352 +0x2d7
main.main.func2(0x3bf36e0, 0xc0004d8800)
	/tmp/build/80754af9/git-gardener.autoscaler-machine-controller-manager-provider.machine-controller-manager-provider/tmp/src/k8s.io/autoscaler/cluster-autoscaler/main.go:431 +0x2a
created by k8s.io/autoscaler/cluster-autoscaler/vendor/k8s.io/client-go/tools/leaderelection.(*LeaderElector).Run
	/tmp/build/80754af9/git-gardener.autoscaler-machine-controller-manager-provider.machine-controller-manager-provider/tmp/src/k8s.io/autoscaler/cluster-autoscaler/vendor/k8s.io/client-go/tools/leaderelection/leaderelection.go:208 +0x113
```

**What you expected to happen**:

No panic

**How to reproduce it (as minimally and precisely as possible)**:

Not clear for now.

**Anything else we need to know**:

**Environment**:
Image: `eu.gcr.io/gardener-project/gardener/autoscaler/cluster-autoscaler:v0.16.0`

--- a/cluster-autoscaler/cloudprovider/mcm/mcm_manager.go
+++ b/cluster-autoscaler/cloudprovider/mcm/mcm_manager.go
@@ -628,7 +628,10 @@ func (m *McmManager) GetMachineDeploymentNodeTemplate(machinedeployment *Machine
 		if err != nil {
 			return nil, fmt.Errorf("Unable to fetch AWSMachineClass object %s, Error: %v", machineClass.Name, err)
 		}
-		awsInstance := aws.InstanceTypes[mc.Spec.MachineType]
+		awsInstance, exists := aws.InstanceTypes[mc.Spec.MachineType]
+		if !exists {
+			return nil, fmt.Errorf("Unable to fetch details for VM type %s", mc.Spec.MachineType)
+		}
 		instance = instanceType{
 			InstanceType: awsInstance.InstanceType,
 			VCPU:         awsInstance.VCPU,
@@ -642,7 +645,10 @@ func (m *McmManager) GetMachineDeploymentNodeTemplate(machinedeployment *Machine
 		if err != nil {
 			return nil, fmt.Errorf("Unable to fetch AzureMachineClass object %s, Error: %v", machineClass.Name, err)
 		}
-		azureInstance := azure.InstanceTypes[mc.Spec.Properties.HardwareProfile.VMSize]
+		azureInstance, exists := azure.InstanceTypes[mc.Spec.Properties.HardwareProfile.VMSize]
+		if !exists {
+			return nil, fmt.Errorf("Unable to fetch details for VM type %s", mc.Spec.Properties.HardwareProfile.VMSize)
+		}
 		instance = instanceType{
 			InstanceType: azureInstance.InstanceType,
 			VCPU:         azureInstance.VCPU,
@@ -666,7 +672,10 @@ func (m *McmManager) GetMachineDeploymentNodeTemplate(machinedeployment *Machine
 				return nil, fmt.Errorf("Unable to convert from %s to %s for %s, Error: %v", kindMachineClass, providerAWS, machinedeployment.Name, err)
 			}
 
-			awsInstance := aws.InstanceTypes[providerSpec.MachineType]
+			awsInstance, exists := aws.InstanceTypes[providerSpec.MachineType]
+			if !exists {
+				return nil, fmt.Errorf("Unable to fetch details for VM type %s", providerSpec.MachineType)
+			}
 			instance = instanceType{
 				InstanceType: awsInstance.InstanceType,
 				VCPU:         awsInstance.VCPU,
@@ -681,8 +690,10 @@ func (m *McmManager) GetMachineDeploymentNodeTemplate(machinedeployment *Machine
 			if err != nil {
 				return nil, fmt.Errorf("Unable to convert from %s to %s for %s, Error: %v", kindMachineClass, providerAzure, machinedeployment.Name, err)
 			}
-
-			azureInstance := aws.InstanceTypes[providerSpec.Properties.HardwareProfile.VMSize]
+			azureInstance, exists := azure.InstanceTypes[providerSpec.Properties.HardwareProfile.VMSize]
+			if !exists {
+				return nil, fmt.Errorf("Unable to fetch details for VM type %s", providerSpec.Properties.HardwareProfile.VMSize)
+			}
 			instance = instanceType{
 				InstanceType: azureInstance.InstanceType,
 				VCPU:         azureInstance.VCPU,

